autoload/ale/fixers/black.vim

```diff
@@ -4,22 +4,32 @@
 call ale#Set('python_black_executable', 'black')
 call ale#Set('python_black_use_global', get(g:, 'ale_use_global_executables', 0))
 call ale#Set('python_black_options', '')
+call ale#Set('python_black_auto_pipenv', 0)
+
+function! ale#fixers#black#GetExecutable(buffer) abort
+    if (ale#Var(a:buffer, 'python_auto_pipenv') || ale#Var(a:buffer, 'python_black_auto_pipenv'))
+    \ && ale#python#PipenvPresent(a:buffer)
+        return 'pipenv'
+    endif
+
+    return ale#python#FindExecutable(a:buffer, 'python_black', ['black'])
+endfunction
 
 function! ale#fixers#black#Fix(buffer) abort
-    let l:executable = ale#python#FindExecutable(
-    \   a:buffer,
-    \   'python_black',
-    \   ['black'],
-    \)
+    let l:executable = ale#fixers#black#GetExecutable(a:buffer)
 
     if !executable(l:executable)
```

Comment:
The test fails because of this check here. I think we can just remove it.

Reply:
Tests are now failing due to this as 0 was an expected case.